sys/drv/pit.c

```diff
@@ -11,56 +11,75 @@ typedef struct pit_state {
   resource_t *regs;
   resource_t *irq_res;
   timer_t timer;
-  uint16_t period_cntr;      /* period as PIT counter value */
-  uint16_t cntr16_prev_read; /* last read counter value */
-  timercntr_t cntr64;        /* counter value since timer initialization */
+  bool overflowed;       /* should we add the counter period */
```

Comment:
I think `overflowed` is a poor name.
As I understand it, `overflowed == true` iff a timer overflow **hasn't** been detected since the last PIT interrupt, which makes it very unintuitive.

Reply:
Renamed it, to `noticed_overflow`, hope now it is intuitive.